src/Sema.zig

```diff
@@ -31539,15 +31539,36 @@ fn elemPtrType(sema: *Sema, ptr_ty: Type, offset: ?usize) !Type {
     const ptr_info = ptr_ty.ptrInfo().data;
     const elem_ty = ptr_ty.elemType2();
     const allow_zero = ptr_info.@"allowzero" and (offset orelse 0) == 0;
+    const target = sema.mod.getTarget();
+    const parent_ty = ptr_ty.childType();
+
+    const vector_info: struct {
+        host_size: u16,
+        bit_offset: u16,
+        alignment: u32,
+    } = if (parent_ty.tag() == .vector) blk: {
+        const elem_bits = elem_ty.bitSize(target);
+        const is_packed = elem_bits != 0 and (elem_bits & (elem_bits - 1)) != 0;
+        // TODO: runtime-known index
+        assert(!is_packed or offset != null);
```

Comment:
I don't think this closes the issue if doing the operation at runtime causes an assertion failure. If this cannot be implemented without knowing the offset then it should be a compile error.

Reply:
If I use the "sema.fail" function to return an error, the parameters "block" and "src" are probably not available, 😭

Reply:
You can always add them if needed.